### ISSUE TITLE
Ch. 2.2: wording improvement suggestion

### ISSUE THREAD

**Issue by PeterCon** [progit/progit2#738]
This paragraph in ch. 2.2 has issues:

> Remember that each file in your working directory can be in one of two states: tracked or untracked.
> Tracked files are files that were in the last snapshot; they can be unmodified, modified, or staged. 
> Untracked files are everything else – any files in your working directory that were not in your last 
> snapshot and are not in your staging area. When you first clone a repository, all of your files will be 
> tracked and unmodified because Git just checked them out and you haven’t edited anything.

First, "can be in one of two states" is immediately contradicted by describing five states: tracked, untracked, unmodified, modified and staged. And the visual just below clearly shows four states. Of course, there are two orthogonal axes involved here.

Suggested revision: "... can be in one of two states in relation to Git source control: tracked or untracked. If a file is untracked, then Git does not have it in any snapshot and is not tracking any changes to it."

Secondly, this statement is not quite correct: "Tracked files are files that were in the last snapshot..." The following sentence ("Untracked files are...") defines the complement set using logical AND, and so this set must be defined with a logical OR. The logical structure of the statement is "property A implies property B", whereas a correct statement of that form would have to be "property A implies (property B OR property C)". If it were flipped around (prop B implies prop A), then that would be a logically-correct statement: "Files that are in unmodified, modified or staged states are tracked." But it wouldn't work here since it assumes familiarity with the not-yet-introduced concepts of unmodified, etc. (There'd also actually be some circularity in explanation.) 

Finally, the opening "Remember" seems out of place. The word "tracked" is used in ch. 2.1, but a contrast with untracked is not mentioned.

So, here's a suggested revision of the paragraph:

> Each file in your working directory is either being tracked by Git, or is untracked. If a file is untracked, Git is aware it is in the working directory, but is basically ignoring it: it's not tracking any changes to it and is not including it in any snapshot. Tracked files are files that Git is monitoring in source control operations, and are always in one of three states: unmodified, modified, or staged. In your working directory, the tracked files will include files that were in the last snapshot. When you first clone a repository, all of your files will be tracked and unmodified because Git just checked them out and you haven’t edited anything. The tracked files will also include any files that previously were untracked but have since been added in your staging area.

**Comment by ivbaoxing** [comment]
Help me to find more with my story

**Comment by ben** [comment]
The bottom line here is that files can be in [_lots_ of states](https://git-scm.com/docs/git-status#_short_format): {unmodified, modified, added, deleted, renamed, copied, updated but unmerged} &times; {working directory, index}. But we don't want to dump all that on the reader at once, we're only in chapter 2, so we write in simpler terms. Hence the original wording: two basic states, where the second one has a couple of substates.

Why don't you open a PR with your changes, and we'll discuss it there?

**Comment by HonkingGoose** [comment]
@ben What are your thoughts on this now? I still like the text that we have in progit2/master currently. I agree that we shouldn't dump everything on the reader at once. I agree that two basic states, and the second state has substates is the right simplification here.

---

@PeterCon is right that this line isn't 100% correct, because a newly created file isn't in any snapshot yet.
>Tracked files are files that were in the last snapshot; they can be unmodified, modified, or staged.

Observe:
1. Create new file
1. ```git add file```
1. file is staged but *not in any snapshot*
1. ```git commit file```
1. file is in snapshot now

However the very next line from the book states:
>Untracked files are everything else — **any files in your working directory that were not in your last snapshot** and are not in your staging area. 
When you first clone a repository, all of your files will be tracked and unmodified because Git just checked them out and you haven’t edited anything.

---

So overall, I'm not sure this actually needs work, I think the proposed rewrite is actually too complicated for a newcomer, even I struggle a bit with it. Also the style of the rewrite is a bit too much "passive voice" for me.

**Comment by ben** [comment]
Failing a PR that actually improves the reader experience here, I'm going to close this out.
